### PR TITLE
Families Bugfixes + Tweaks: Even Distribution of Leaders, Fixes HoS-Family-Leaders, Adds Admin Tool to make Leaders

### PR DESCRIPTION
**Pull request by Iamgoofball**
## About The Pull Request

Leaders are now distributed evenly among the Families.
As a result, the Families rulesets now only require 3 people to start, instead of 9.
You can no longer become a Family Leader as a restricted role, such as a Head of Security.
Admins can now promote people to Family Leader with a button from the traitor panel.

## Why It's Good For The Game

Bugfixes are good. Better distribution of family leaders is also good.

## Changelog

:cl:
qol: Leaders are now distributed evenly among the Families.
qol: As a result, the Families rulesets now only require 3 people to start, instead of 9.
fix: You can no longer become a Family Leader as a restricted role, such as a Head of Security.
admin: Admins can now promote people to Family Leader with a button from the traitor panel.
/:cl:

### CODE REVIEW

File: code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm [hunk 1]
```diff
@@ -252,7 +252,7 @@
 	antag_flag_override = ROLE_FAMILIES
 	protected_roles = list("Prisoner", "Head of Personnel")
 	restricted_roles = list("Cyborg", "AI", "Security Officer", "Warden", "Detective", "Head of Security", "Captain", "Research Director")
-	required_candidates = 9
+	required_candidates = 0
```
Comment (Ghommie) [inline code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm:255]: Didn't you say it requires 3 candidates now?

File: code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm [hunk 2]
```diff
@@ -496,7 +496,7 @@
 	antag_flag = ROLE_FAMILIES
 	protected_roles = list("Prisoner", "Head of Personnel")
 	restricted_roles = list("Cyborg", "AI", "Security Officer", "Warden", "Detective", "Head of Security", "Captain", "Research Director")
-	required_candidates = 9
+	required_candidates = 0
```
Comment (Ghommie) [inline code/game/gamemodes/dynamic/dynamic_rulesets_roundstart.dm:499]: Ditto.

File: code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm (hunk 1 above)
Comment (Iamgoofball) [reply code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm:255]: I FORGOT TO PUSH

**Comment by Draggeru** [comment]
"Ah yes, everyone has turned families off so I will lower the player requirement so the majority of people who don't want to play families still have to play in gang rounds."

**Comment by CMDR-Gungnir** [comment]
I love the mode.

But after how every Families round has gone lately on Manuel, I don't think it's working here, man. Unfortunately, the lowest common denominator of human is an unwashed bonobo and that seems to be what the Free Antag brings out of people.

**Comment by Iamgoofball** [comment]
github please

**Pull request merged**